app_dart/lib/src/request_handlers/reset_luci_task.dart

```diff
@@ -0,0 +1,62 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:appengine/appengine.dart';
+import 'package:cocoon_service/cocoon_service.dart';
+import 'package:gcloud/db.dart';
+import 'package:meta/meta.dart';
+
+import '../datastore/cocoon_config.dart';
+import '../model/appengine/key_helper.dart';
+import '../model/appengine/task.dart';
+import '../request_handling/api_request_handler.dart';
+import '../request_handling/authentication.dart';
+import '../request_handling/body.dart';
+import '../request_handling/exceptions.dart';
+import '../service/datastore.dart';
+
+@immutable
+class ResetDevicelabTask extends ApiRequestHandler<Body> {
+  const ResetDevicelabTask(
+    Config config,
+    AuthenticationProvider authenticationProvider, {
+    @visibleForTesting DatastoreServiceProvider datastoreProvider,
+    @required this.buildBucketClient,
+  })  : datastoreProvider =
+            datastoreProvider ?? DatastoreService.defaultProvider,
+        super(config: config, authenticationProvider: authenticationProvider);
+
+  final DatastoreServiceProvider datastoreProvider;
+  final BuildBucketClient buildBucketClient;
+
+  static const String keyParam = 'Key';
+
+  @override
+  Future<Body> post() async {
+    checkRequiredParameters(<String>[keyParam]);
+    final DatastoreService datastore = datastoreProvider(config.db);
+    final String encodedKey = requestData[keyParam] as String;
+    final ClientContext clientContext = authContext.clientContext;
+    final KeyHelper keyHelper =
+        KeyHelper(applicationContext: clientContext.applicationContext);
+    final Key key = keyHelper.decode(encodedKey);
+
+    final LuciBuildService luciBuildService = LuciBuildService(
+      config: config,
+      buildBucketClient: buildBucketClient,
+    );
+    // TODO
+    await luciBuildService.rescheduleBuild(
```

Comment:
I'm looking for the best way to get the build info to schedule a rerun, without routing it through the front end.

@keyonghan does the backend store the luci build info anywhere today?

Reply:
AFAIK, backend only stores luci build info when refreshing new github commits: https://github.com/flutter/cocoon/blob/master/app_dart/lib/src/request_handlers/refresh_github_commits.dart#L218

Reply:
Then refresh status via: https://github.com/flutter/cocoon/blob/master/app_dart/lib/src/request_handlers/refresh_chromebot_status.dart

Reply:
Does the LuciTask object have this info? I could lookup the recent ones via https://github.com/flutter/cocoon/blob/master/app_dart/lib/src/request_handlers/refresh_chromebot_status.dart#L53 somehow?

Alternatively, can this info be stored on an entity linked to the task id?

Reply:
Right now, we do not save LuciTask in datastore, but definitely we can create a new entity to save this info. if this can make it easier.

Reply:
If the API to look them up from LUCI is fast enough, it seems like it would be better to not cache them to avoid filling up datastore. WDUT?

Reply:
I would use the current API call first before saving them to datastore. On the other hand, reset-luci doesn't seem to be called that frequently if implemented.

Reply:
Updated this - based on my understanding of the API, I think I just need to schedule a new build? I think luci_status would pick it up again If I understand how it works

Reply:
I guess so. but one thing to make sure is whether we just need the latest status of one SHA (based on the assumption that the returned status is the latest), or we want to maintain all statuses. If the latter case, the frontend may consider to present them all, similar to the devicelab task.